File: src/main/java/at/xirado/bean/data/BasicAutocompletionChoice.java
package at.xirado.bean.data;

import net.dv8tion.jda.api.interactions.commands.Command;

public class BasicAutocompletionChoice implements IAutocompleteChoice
{
    private final String name, value;

    public BasicAutocompletionChoice(String name, String value)
    {
        this.name = name;
        this.value = value;
    }

    @Override
    public Command.Choice toCommandAutocompleteChoice()
    {
        return new Command.Choice(name, value);
    }

    @Override
    public String getName()
    {
        return name;
    }

    @Override
    public String getValue()
    {
        return value;
    }
}

File: src/main/java/at/xirado/bean/data/BasicAutoCompletionChoice.kt
package at.xirado.bean.data

import net.dv8tion.jda.api.interactions.commands.Command

class BasicAutoCompletionChoice(
    override val name: String,
    override val value: String
) : IAutoCompleteChoice {
    override fun toChoice() = Command.Choice(name, value)
}
